.golangci.yaml

```diff
@@ -0,0 +1,59 @@
+linters:
+  enable:
+    - deadcode
+    - errcheck
+    - gosimple
+    - govet
+    - ineffassign
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+    - asciicheck
+    - bodyclose
+    - depguard
+    - dogsled
+    - dupl
+    - exhaustive
+    - exportloopref
+    - funlen
+    - gochecknoinits
+    - gocognit
+    - goconst
+    - gocritic
+    - gocyclo
+    - godot
+    - gofmt
+    - gofumpt
+    - goheader
+    - goimports
+    - golint
+    - gomnd
+    - gomodguard
+    - goprintffuncname
+    - gosec
+    - interfacer
+    - maligned
+    - misspell
+    - nakedret
+    - nestif
+    - noctx
+    - nolintlint
+    - prealloc
+    - rowserrcheck
+    - scopelint
+    - sqlclosecheck
+    - stylecheck
+    - unconvert
+    - unparam
+    - whitespace
+  disable:
+    - gci
```

Comment:
This one requires that imports be in a certain order. In my most recent PR I made a silent decision (sorry) to have three import blocks: stdlib, third party and module-local. gci is meant to allow that (according to its docs) but it didnt work.

I now have a script in my bindir called `gofumports-local` that contains:
```
#!/bin/sh
exec gofumports -local $(go list -m) "$@"
```

It seems there is no setting where gofumports figures out for itself what the current module is, so you have to tell it. I have my editor set up to run that. This means you probably need to do the same, This is far from ideal so maybe I should just get rid of this three import sections idea and just use gofumports without the `-local` option.

Thoughts?

Reply:
I would opt for doing something more standard, especially if you think others might contribute at some point.

Local imports are still grouped within third party imports (and in our case are usually at the top anyway).

It's not that I disagree with your wish to have a block for local, 3p and stdlib - I just don't think doing something non standard here is worth the effort. But maybe I'm just lazy and I should change this attitude.

I think if we want to proceed the gofumports-local path what we need is a `make fmt` target that does it the way we want it and also runs on CI as part of lint ensuring nothing is updated - we should probably have this target regardless for new comers.

Then I'm happy with it ;)

Reply:
oh and a `make get-tools` target or similar.

Reply:
all good points.